Signal/src/UIView+OWS.h

```diff
@@ -60,6 +60,31 @@ CGFloat ScaleFromIPhone5(CGFloat iPhone5Value);
 
 - (void)centerOnSuperview;
 
+#pragma mark - RTL
+
+// For correct right-to-left layout behavior, use "leading" and "trailing",
+// not "left" and "right".
+//
+// NOTE: the margin values are inverted in RTL layouts.
+- (BOOL)isRTL;
+- (NSArray<NSLayoutConstraint *> *)autoPinLeadingAndTrailingToSuperview;
+- (NSLayoutConstraint *)autoPinLeadingToSuperView;
+- (NSLayoutConstraint *)autoPinLeadingToSuperViewWithMargin:(CGFloat)margin;
+- (NSLayoutConstraint *)autoPinTrailingToSuperView;
+- (NSLayoutConstraint *)autoPinTrailingToSuperViewWithMargin:(CGFloat)margin;
+- (NSLayoutConstraint *)autoPinLeadingToTrailingOfView:(UIView *)view;
+- (NSLayoutConstraint *)autoPinLeadingToTrailingOfView:(UIView *)view margin:(CGFloat)margin;
+- (NSLayoutConstraint *)autoPinLeadingToView:(UIView *)view;
+- (NSLayoutConstraint *)autoPinLeadingToView:(UIView *)view margin:(CGFloat)margin;
+- (NSLayoutConstraint *)autoPinTrailingToView:(UIView *)view;
+- (NSLayoutConstraint *)autoPinTrailingToView:(UIView *)view margin:(CGFloat)margin;
+// Return Right on LTR and Right on RTL.
+- (NSTextAlignment)textAlignmentUnnatural;
```

Comment:
The "default" text alignment is left in LTR and right in RTL.  In some cases you want the opposite.  I've called that "Unnatural" - I couldn't think of a better name.